File: Wirtualny_Swiat_Java/Projekt_PO_Java/src/model/Organisms/Guarana.java
package model.Organisms;

import model.Organism;
import model.Plant;
import model.Point;
import model.World;

import java.awt.*;

public class Guarana extends Plant {
    private final Color guaranaColor = Color.gray;

    public Guarana(World world, Point position){
        super(world, position, 0);
    }

    @Override
    public String getNazwa(){
        return "Guarana";
    }

    @Override
    public Color getColor(){
        return guaranaColor;
    }

    @Override
    public Organism makeNewOrganism(Point position){
        return new Guarana(world, position);
    }
}

File: WirtualnySwiatPython/Plants/Guarana.py
from plant import Plant
from config import CYAN

class Guarana(Plant):
    def __init__(self, world, position):
        super().__init__(world, position, power=0)
        self.color = CYAN

    def getNazwa(self):
        return "Guarana"

    def getColor(self):
        return self.color

    def makeNewOrganism(self, position):
        return Guarana(self.world, position)

    def Collision(self, attacker):
        attacker.setPower(attacker.getPower() + 3)

        self.collisionHelper(attacker, self.getPosition())